### PR TITLE
Teacher Page

### LINKED ISSUE #43: Redesign Teacher Page

### PR DESCRIPTION
**Pull request by LuisDuarte1**
Closes #43 
Screenshot 2023-06-12 at 18-47-22 FEUP - André Restivo

### CODE REVIEW

File: content-scripts/src/modules/utilities/pageUtils.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+                element.remove()
+                div.appendChild(element)
+            })
+            outerElement.appendChild(div)
+            elementList = [row]
+            continue
+        }
+        if (elementList.length === 0) {
+            continue
+        }
```
Comment (thePeras) [inline content-scripts/src/modules/utilities/pageUtils.js:24]: You can push this to top and remove inside the ifs verification

File: content-scripts/src/modules/utilities/pageUtils.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        const div = document.createElement("div")
+        elementList.forEach((element) => {
+            element.remove()
+            div.appendChild(element)
+        })
+        outerElement.appendChild(div)
+    }
+}
+
+export function removeElementWithoutChildren(selector){
```
Comment (thePeras) [inline content-scripts/src/modules/utilities/pageUtils.js:37]: Can you think of a better name?

File: content-scripts/src/modules/utilities/pageUtils.js [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+}
+
+
+export function removeTwoColumnTable(tableSelector, inverted=false){
+    const table = document.querySelector(tableSelector);
+    if(table === null || table.tagName !== "TABLE") 
+        throw Error("Couldnt find table with " + tableSelector +  " selector");
+    const tbody = table.children[0];
+    const div = document.createElement("div");
+    div.classList.add("keyPairTable")
```
Comment (thePeras) [inline content-scripts/src/modules/utilities/pageUtils.js:55]: Our own class name start with `se-` and follow a kebab-case convention.

**Review by thePeras — CHANGES_REQUESTED** [review]

File: content-scripts/src/modules/utilities/pageUtils.js [hunk 4]
```diff
@@ -0,0 +1,122 @@
+export function groupSectionTitleAndContent(selector) {
+    //this detects if an element is an h3 and groups 
+    //with the next element until the next h3 appears, if any
+    const outerElement = document.querySelector(selector);
+    var elementList = [];
```
[the remark this answers is not in the document]
Comment (LuisDuarte1) [reply content-scripts/src/modules/utilities/pageUtils.js]: fixed sorry restivo :)

File: content-scripts/src/pages/teacher_page.js [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+
+
+function makePublicationWebsiteButtons() {
+    const tabelasz = document.querySelector(".tabelasz > tbody")
+    const informacaoPessoal = document.querySelector(".informacao-pessoal-dados");
+    const websiteList = document.createElement("div")
+    websiteList.classList.add("se-publication-website-list")
+    const listOfRows = [...tabelasz.children]
+    listOfRows.forEach((row) => {
+        console.log(row)
```
Comment (thePeras) [inline content-scripts/src/pages/teacher_page.js]: remove this print

File: content-scripts/src/pages/teacher_page.js [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+        const linkElement = row.querySelector("td:nth-child(2) > a")
+        if (linkElement === null) {
+            return;
+        }
+        const link = linkElement.href
+        console.log(link)
```
Comment (thePeras) [inline content-scripts/src/pages/teacher_page.js]: Remove this print

**Review by thePeras — CHANGES_REQUESTED** [review]
Change divs to buttons for better acessebility

**Comment by ttoino** [comment]
> Change divs to buttons for better acessebility

I think an anchor element would be more appropriate, we're just opening a new page after all

**Comment by toni-santos** [comment]
> > Change divs to buttons for better acessebility
> 
> I think an anchor element would be more appropriate, we're just opening a new page after all

I agree with this, an example as to why would be the case where we attempt to open a link in a new tab with a button like Mouse-3 or Ctrl + Mouse-1, the expected behavior is for a new tab to open but currently it doesn't work, in the case of Mouse-3, or it opens in the same tab, in the case of Ctrl + Mouse-1.

Other than that this is looking great! :)

**Comment by LuisDuarte1** [comment]
You were completely right, this was written on a late night so I didn't bother then ahah, nice catch!

**Review by thePeras — APPROVED** [review]
Perfect!
✅ 🛫

**Pull request merged**